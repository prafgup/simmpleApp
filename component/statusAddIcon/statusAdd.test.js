import React from "react";
import {shallow} from "enzyme";

import AddStatusIcon from "./index";


describe("AddStatusIcon",()=>{

    describe("AddStatusIcon",()=>{
        let component ;

        beforeEach(()=>{
            component = shallow(<AddStatusIcon/>);
        });

        it("Should match to snapshot",()=>{
            expect(component).toMatchSnapshot("Add Status Icon");
        });

        it("Should Render all Plus Icon",()=>{
            expect(component.find("[data-test='icon']").length).toBe(1);
        });
        it("Should Render all Whole Container",()=>{
            expect(component.find("[data-test='statusAdd']").length).toBe(1);
        });
    })
});