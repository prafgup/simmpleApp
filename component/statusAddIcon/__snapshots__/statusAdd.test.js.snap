// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStatusIcon AddStatusIcon Should match to snapshot: Add Status Icon 1`] = `
AddStatusIcon {
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <AddStatusIcon />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <View
        data-test="statusAdd"
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": 12.5,
              "height": 25,
              "width": 25,
            }
          }
        />
        <View
          style={
            Object {
              "left": 6,
              "position": "relative",
              "top": -20,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="white"
            data-test="icon"
            name="plus"
            size={17}
          />
        </View>
      </View>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
